weed/filer/filer_delete_entry.go

```diff
@@ -49,13 +49,12 @@ func (f *Filer) DeleteEntryMetaAndData(ctx context.Context, p util.FullPath, isR
 		return fmt.Errorf("delete file %s: %v", p, err)
 	}
 
-	if shouldDeleteChunks && !isDeleteCollection {
-		f.DirectDeleteChunks(entry.GetChunks())
-	}
-
-	if isDeleteCollection {
-		collectionName := entry.Name()
-		f.doDeleteCollection(collectionName)
+	if shouldDeleteChunks {
+		if isDeleteCollection {
```

Comment:
This piece of logic is tricky. A unit test would be really nice.

Reply:
> This piece of logic is tricky. A unit test would be really nice.
I saw that in order to do a unit test you need to initialize the Filer structure, which is not a completely trivial task.

Here the logic is simple: shouldDeleteChunks is a synonym for fs.mv(move), so chunks do not need to be deleted, including the entire collection.